src/browser-realm.ts

```diff
@@ -191,46 +65,26 @@ function removeIframe(iframe: HTMLIFrameElement) {
     }
 }
 
-
 export default function createSecureEnvironment(distortionMap?: Map<RedProxyTarget, RedProxyTarget>, endowments?: object): (sourceText: string) => void {
     const iframe = createDetachableIframe();
-
-    // For Chrome we evaluate the `window` object to kickstart the realm so that
-    // `window` persists when the iframe is removed from the document.
-    const redGlobalThis = (iframe.contentWindow as WindowProxy).window;
-    const { eval: redIndirectEval } = redGlobalThis;
-    redIndirectEval('window');
-    const blueGlobalThis = globalThis;
-
-    removeIframe(iframe);
-
-    const blueRefs = getCachedReferences(blueGlobalThis);
-    const redRefs = getCachedReferences(redGlobalThis);
-
+    const blueWindow = window;
+    const redWindow = (iframe.contentWindow as WindowProxy).window;
+    const endowmentsDescriptors = getFilteredEndowmentDescriptors(endowments || {});
+    const { eval: redIndirectEval } = redWindow;
+    // extra the global references and descriptors before any interference
+    const blueRefs = getCachedReferences(blueWindow);
+    const redRefs = getCachedReferences(redWindow);
+    // creating a new environment
     const env = new SecureEnvironment({
-        blueGlobalThis,
-        redGlobalThis,
+        blueGlobalThis: blueWindow,
+        redGlobalThis: redWindow,
         distortionMap,
     });
-
-    const globalDescriptors = aggregateGlobalDescriptors(
-        redRefs.windowDescriptors,
-        blueRefs.windowDescriptors,
-        endowments && getOwnPropertyDescriptors(endowments)
-    );
-
-    // remapping globals
-    env.remap(redRefs.window, blueRefs.window, globalDescriptors);
-
-    // remapping unforgeable objects
-    env.remap(redRefs.document, blueRefs.document); // no descriptors needed, document.location is unforgeable
-    env.remap(redRefs.EventTargetProto, blueRefs.EventTargetProto, blueRefs.EventTargetProtoDescriptors);
-    env.remap(redRefs.WindowPropertiesProto, blueRefs.WindowPropertiesProto, blueRefs.WindowPropertiesProtoDescriptors);
-    env.remap(redRefs.WindowProto, blueRefs.WindowProto, blueRefs.WindowProtoDescriptors);
-
-    // adjusting proto chains when possible
-    ReflectSetPrototypeOf(redRefs.document, env.getRedValue(blueRefs.DocumentProto));
-
+    linkIntrinsics(env, blueWindow, redWindow);
```

Comment:
abstracting all the operations into methods that can be understood by just reading them